BigQuery/tests/Unit/JsonTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery\Tests\Unit;
+
+use Google\Cloud\BigQuery\Json;
+use InvalidArgumentException;
+use JsonSerializable;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group bigquery
+ * @group bigquery-json
+ */
+class JsonTest extends TestCase
+{
+    public function testInvalidValues()
```

Comment:
This tests only one type of invalid value, you need to add more, probably from a data provider.

Reply:
I could only think of couple of more, added them.